src/main/kotlin/no/fdk/fdk_reasoning_service/config/CircuitBreakerConsumerConfiguration.kt

```diff
@@ -0,0 +1,51 @@
+package no.fdk.fdk_reasoning_service.config
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker.StateTransition
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry
+import io.github.resilience4j.circuitbreaker.event.CircuitBreakerOnStateTransitionEvent
+import no.fdk.fdk_reasoning_service.kafka.KafkaManager
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+open class CircuitBreakerConsumerConfiguration(
+    private val circuitBreakerRegistry: CircuitBreakerRegistry,
+    private val kafkaManager: KafkaManager
+) {
+    init {
+        LOGGER.debug("Configuring circuit breaker event listener")
+        val topics = listOf(
+            "dataset-reasoning",
+            "concept-reasoning",
+            "data-service-reasoning",
+            "information-model-reasoning",
+            "service-reasoning",
+            "event-reasoning"
+        )
+        topics.forEach {
+            circuitBreakerRegistry.circuitBreaker(it).eventPublisher.onStateTransition { event: CircuitBreakerOnStateTransitionEvent ->
+                handleStateTransition(event, it)
+            }
+        }
```

Comment:
det var ikke lista med topics som var problemet her, det er den lange linja med mye som skjer. Og it blir litt for generisk når det mye som skjer på en gang, bruk mer forklarende variabel:
```suggestion
        topics.forEach { topic ->
            circuitBreakerRegistry.circuitBreaker(topic)
              .eventPublisher
              .onStateTransition { event: CircuitBreakerOnStateTransitionEvent ->
                handleStateTransition(event, topic)
            }
        }
```